control_plane/attachment_service/src/service.rs

```diff
@@ -742,42 +745,272 @@ impl Service {
             (waiters, response_shards)
         };
 
-        let deadline = Instant::now().checked_add(Duration::from_secs(5)).unwrap();
+        self.await_waiters(waiters).await?;
+
+        Ok(TenantCreateResponse {
+            shards: response_shards,
+        })
+    }
+
+    /// Helper for functions that reconcile a number of shards, and would like to do a timeout-bounded
+    /// wait for reconciliation to complete before responding.
+    async fn await_waiters(
+        &self,
+        waiters: Vec<ReconcilerWaiter>,
+    ) -> Result<(), ReconcileWaitError> {
+        let deadline = Instant::now().checked_add(Duration::from_secs(30)).unwrap();
         for waiter in waiters {
             let timeout = deadline.duration_since(Instant::now());
             waiter.wait_timeout(timeout).await?;
         }
-        Ok(TenantCreateResponse {
-            shards: response_shards,
-        })
+
+        Ok(())
     }
 
-    pub(crate) async fn tenant_timeline_create(
+    /// This API is used by the cloud control plane to do coarse-grained control of tenants:
+    /// - Call with mode Attached* to upsert the tenant
+    /// - Call
+    pub(crate) async fn tenant_location_config(
         &self,
         tenant_id: TenantId,
-        mut create_req: TimelineCreateRequest,
-    ) -> Result<TimelineInfo, ApiError> {
-        let mut timeline_info = None;
+        req: TenantLocationConfigRequest,
```

Comment:
This request contains a tenant_shard_id field. Doesn't make sense anymore, does it? When will we clean that up?

Reply:
Much later: when we eventually move to a model where all tenants are managed via this service, the control plane can call a simpler API for loading a tenant, rather than this "virtual pageserver" API.  At that point, this API reverts to being a pure pageserver API where the shard always makes sense.

(or if we eventually move to a model where the control plane just knows how to do sharding itself, same result)